raiden/connection_manager.py

```diff
@@ -126,34 +143,48 @@ def leave(self, wait_for_settle=True, timeout=30):
                 c.partner_address) for c in open_channels]
             for channel in channel_specs:
                 try:
-                    self.api.close(*channel),
-                except RuntimeError:
+                    self.api.close(*channel)
+                except:
                     # if the error wasn't that the channel was already closed: raise
                     if channel[1] in [c.partner_address for c in self.open_channels]:
                         raise
 
             # wait for events to propagate
             gevent.sleep(self.raiden.alarm.wait_time)
 
-            if wait_for_settle:
-                try:
-                    with gevent.timeout.Timeout(timeout):
-                        while any(c.state != CHANNEL_STATE_SETTLED for c in open_channels):
-                            # wait for events to propagate
-                            gevent.sleep(self.raiden.alarm.wait_time)
-
-                except gevent.timeout.Timeout:
-                    log.debug(
-                        'timeout while waiting for settlement',
-                        unsettled=sum(
-                            1 for channel in open_channels if
-                            channel.state != CHANNEL_STATE_SETTLED
-                        ),
-                        settled=sum(
-                            1 for channel in open_channels if
-                            channel.state == CHANNEL_STATE_SETTLED
-                        )
+        if wait_for_settle:
```

Comment:
Also, with the introduction of the AsyncResult what is the point of the `wait_for_settle`? One can just call the function async and not wait, or wait for it to properly leave the network by waiting for the async. (Also the flag is weird because it's basically an implicit function call and we have no requirements for backwards compatibility at this point)